readme.md

```diff
@@ -3,8 +3,53 @@
 This package contains a parser for the Checker Framework's stub files:
 https://checkerframework.org/manual/#stub
 It is a fork of the [JavaParser](http://javaparser.org) project.
-The remainder of this README file is the JavaParser README.
 
+## Differences between the StubParser and JavaParser
+
+Following list is the differences that were added to the JavaParser to adjust it for using as Stubparser for [the Checker Framework](https://github.com/typetools/checker-framework).
+
+1. StubUnit class that represents the parsed [stubfile](https://checkerframework.org/manual/#stub).
+2. Changes at the java.jj file to parse the stubfiles.
+3. Methods for parsing the stubfiles at the JavaParser.class.
+
+## Updating from upstream JavaParser
+
+This section describes how to incorporate changes from JavaParser into
+StubParser.  Only developers, not users, of StubParser need to do this.
+
+1. Fork [the stubparser project](https://github.com/typetools/stubparser) to your github account.
+2. Enable Travis build for your fork of the stubparser.
+3. Clone the repository.
+```bash
+git clone https://github.com/{user.name}/stubparser
+```
+4. Enter to the main directory of the local project.
+```bash
+cd stubparser
+```
+5. Create and checkout a branch named `updating`.
+```bash
+git checkout -b updating
+```
+6. Pull the upstream of [the JavaParser project](https://github.com/javaparser/javaparser).
+```bash
+git pull https://github.com/javaparser/javaparser master
+```
+7. Resolve conflicts if required and commit it.
+8. Run maven tests in the root stubparser directory.  If any tests fail, fix them before continuing.
+```bash
+mvn test
+```
+10. Push commits to your fork of the stubparser.

```

Comment:
This file uses all three of "StubParser", "Stubparser", and "stubparser".  Settle on one for the name of the tool, and use it consistently.  (The directory named `stubparser` shouldn't be capitalized, though.)